commands/addCategory.ts

```diff
@@ -38,46 +38,55 @@ export async function execute(interaction: CommandInteraction) {
   }
 
   await interaction.reply({
-    content: `Are you sure you want to add ${category.trim()}?`,
+    content: `Are you sure you want to add \`${category.trim()}\`?`,
     components: [buttonRow],
     ephemeral: true,
   });
 
-  const buttonReply = await interaction.channel?.awaitMessageComponent({ componentType: 'BUTTON' });
-  if (!buttonReply) {
-    return;
-  }
+  const interactionMessage = await interaction.fetchReply();
 
-  const buttonSelected = buttonReply.customId;
-  buttonReply.update({ components: [] });
-  if (buttonSelected === REPLY.NO) {
-    buttonReply.followUp({
-      content: `"${category.trim()}" was not added`,
-      ephemeral: true,
-    })
-    return;
-  }
-  else {
-    try {
-      const foundCategory = await findCategoryByName(category.trim());
-      if (foundCategory) {
-        await interaction.editReply('This category is already registered.');
-      }
-      else {
-        await createCategory(category.trim());
-        await interaction.editReply('Thank you. The category has been added.');
-      }
+  if (interactionMessage instanceof Message) {
```

Comment:
Minor, but imo the code would be slightly more readable if we returned early instead of nesting the logic. So if `if(!(interactionMessage instanceof Message)) { return; }`

Reply:
I tried that initially and TypeScript still complained, let me try again, I might have missed a type check.

Reply:
OK, so maybe VS Code was acting up the first time I tried this. Now it looks like it is working!
I definitely prefer the early return, thanks for making me double check.